### PR TITLE
FEAT: Add `n_images` to `ImageProperties`

### PR DESCRIPTION
**Pull request by lschr**
This exposes the number of frames/images in a file via `improps()`.

See also #915.

### CODE REVIEW

File: imageio/core/v3_plugin_api.py [hunk 1]
```diff
@@ -35,6 +37,7 @@ class ImageProperties:
 
     shape: Tuple[int, ...]
     dtype: np.dtype
+    n_images: Optional[int]
```
Comment (FirefoxMetzger) [inline imageio/core/v3_plugin_api.py:40]:
```suggestion
    n_images: Optional[int] = None
```

Saves us from setting `None` when reading single images.

File: imageio/core/legacy_plugin_wrapper.py [hunk 2]
```diff
@@ -291,12 +291,22 @@ def properties(self, index=None):
...
+                n_images=n_images,
+                is_batch=True,
+            )
 
+        image = self.read(index=index)
         return ImageProperties(
             shape=image.shape,
             dtype=image.dtype,
-            is_batch=True if index is None else False,
+            n_images=None,
```
Comment (FirefoxMetzger) [inline imageio/core/legacy_plugin_wrapper.py:308]:
```suggestion
```

We can make `None` the default in the definition.

File: imageio/core/v3_plugin_api.py [hunk 3]
```diff
@@ -20,6 +20,8 @@ class ImageProperties:
         The shape of the loaded ndimage.
     dtype : np.dtype
         The dtype of the loaded ndimage.
+    n_images : int or None
```
Comment (FirefoxMetzger) [inline imageio/core/v3_plugin_api.py:23]:
```suggestion
    n_images : int
```

We don't document `None` if it is the default value.

File: imageio/plugins/tifffile_v3.py [hunk 4]
```diff
@@ -368,13 +371,24 @@ def properties(self, *, index: int = None, page: int = None) -> ImageProperties:
...
+                dtype=target_page.dtype,
+                n_images=n_pages,
                 is_batch=True,
                 spacing=_get_resolution(target_page)["resolution"],
             )
         else:
             props = ImageProperties(
                 shape=target_page.shape,
                 dtype=target_page.dtype,
+                n_images=None,
```
Comment (FirefoxMetzger) [inline imageio/plugins/tifffile_v3.py:391]:
```suggestion
```

File: imageio/plugins/opencv.py [hunk 5]
```diff
@@ -270,13 +270,28 @@ def properties(
...
+                n_images=n_images,
+                is_batch=True,
+            )
 
+        img = self.read(index=index, flags=flags, colorspace=colorspace)
         return ImageProperties(
-            shape=img.shape,
-            dtype=img.dtype,
-            is_batch=(index is ...),
+            shape=img.shape, dtype=img.dtype, n_images=None, is_batch=False
```
Comment (FirefoxMetzger) [inline imageio/plugins/opencv.py:294]:
```suggestion
            shape=img.shape, dtype=img.dtype, is_batch=False
```

**Review by FirefoxMetzger — APPROVED** [review]
Nice work, the code is really clean and I only have a few minor comments. Thanks for the PR 🚀 

Regarding the CI failure: Is this branch in sync with the latest master? It failed because of an old issue that I've fixed a few days ago. If it is on the latest version and still fails I will have to reopen that issue.

**Comment by lschr** [comment]
> Regarding the CI failure: Is this branch in sync with the latest master?

I think this was a legitimate complaint. I had forgotten to update OpenCV tests, and there was no test for calling `properties()` with integer index.

Also, I vaguely remember some inconsistencies in the documentation of `properties()`, `metadata()` and `read()` regarding `index=None` vs `index=...`. Is this material for another PR or should we fix it here? (Or maybe my memory is wrong and there are no problems.)

**Comment by FirefoxMetzger** [comment]
> Also, I vaguely remember some inconsistencies in the documentation of properties(), metadata() and read() regarding index=None vs index=.... Is this material for another PR or should we fix it here? (Or maybe my memory is wrong and there are no problems.)

That could be the case. In the initial iterations of v3 we used `None` to indicate "read all ndimages or file-level metadata". However, we switched this to `...` soon after, because we needed a sentinel to indicate "no default / plugin-level default" and it made sense to have that be `None`, which is typically used for this purpose.

Let's fix any inconsistencies in a separate PR. This one is ready to go and can be part of Monday's release :)

File: imageio/core/v3_plugin_api.py [hunk 6]
```diff
@@ -20,6 +20,8 @@ class ImageProperties:
         The shape of the loaded ndimage.
     dtype : np.dtype
         The dtype of the loaded ndimage.
+    n_images : int
+        Number of images in the file if ``index=...``, `None` for single images
```
Comment (FirefoxMetzger) [inline imageio/core/v3_plugin_api.py:24]:
```suggestion
        Number of images in the file if ``index=...``, `None` for single images.
```

**Pull request merged**